### PR TITLE
Update CA API repo and incorporate API changes in Sidecar resource

### PR DESCRIPTION
**Pull request by myidpt**
This PR is part of to the API definition move: https://github.com/istio/api/pull/1520

Also merging https://github.com/istio/istio/pull/25585 together to resolve conflicts.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
I think we will run into conflicts with some other changes from API that also haven't been merged ihn https://github.com/istio/istio/pull/25585/files

**Comment by myidpt** [comment]
@howardjohn I see.
@nrjpoddar I can merge your change https://github.com/istio/istio/pull/25585 to here. Or if you prefer, you can merge my change over to your PR.

**Comment by nrjpoddar** [comment]
@myidpt take my changes and merge here, thanks!

**Review by howardjohn — APPROVED** [review]

**Review by JimmyCYJ — APPROVED** [review]

**Review by nrjpoddar — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: galley/pkg/config/analysis/analyzers/deprecation/deprecation.go [hunk 1]
```diff
@@ -60,29 +60,6 @@ func (*FieldAnalyzer) analyzeSidecar(r *resource.Instance, ctx analysis.Context)
 
 	sc := r.Message.(*v1alpha3.Sidecar)
 
-	if sc.Localhost != nil {
```
Comment (therealmitchconnors) [inline galley/pkg/config/analysis/analyzers/deprecation/deprecation.go:63]: Can you explain why these messages are being removed?  I don't see any API changes to sidecar as a part of this PR, but maybe I'm missing something?

Comment (howardjohn) [reply galley/pkg/config/analysis/analyzers/deprecation/deprecation.go:63]: From https://github.com/istio/api/pull/1529

The API was merged but never implemented or shipped to users (always hide from docs + unimplemented)

Comment (myidpt) [reply galley/pkg/config/analysis/analyzers/deprecation/deprecation.go:63]: @nrjpoddar knows better. I'm copying from https://github.com/istio/istio/pull/25585

**Review by esnible — APPROVED** [review]

**Pull request merged**